api/src/main/java/net/kyori/adventure/text/format/StyleImpl.java

```diff
@@ -117,56 +105,29 @@ static void decorate(final Builder builder, final TextDecoration[] decorations)
 
   @Override
   public TextDecoration.@NotNull State decoration(final @NotNull TextDecoration decoration) {
-    if (decoration == TextDecoration.BOLD) {
-      return this.bold;
-    } else if (decoration == TextDecoration.ITALIC) {
-      return this.italic;
-    } else if (decoration == TextDecoration.UNDERLINED) {
-      return this.underlined;
-    } else if (decoration == TextDecoration.STRIKETHROUGH) {
-      return this.strikethrough;
-    } else if (decoration == TextDecoration.OBFUSCATED) {
-      return this.obfuscated;
+    final @Nullable TextDecoration.State state = this.decorations.get(decoration);
```

Comment:
Same here, do we need to non-null check decoration here?

Reply:
Ignore that, this works as a null-check for the decoration passed (though it could be changed to a reqnn, comments?)

Reply:
Added a comment for now, but I definitely want to get some more feedback on this particular element